src/ucp/core/ucp_mm.c

```diff
@@ -415,6 +417,14 @@ ucs_status_t ucp_mem_map(ucp_context_h context, const ucp_mem_map_params_t *para
         goto out;
     }
 
+    if (((flags & UCP_MEM_MAP_FIXED) || (flags & UCP_MEM_MAP_NONBLOCK)) &&
+        ((params->field_mask & UCP_MEM_MAP_PARAM_FIELD_MEMORY_TYPE)
+         && (params->memory_type != UCS_MEMORY_TYPE_HOST))) {
+        ucs_error("UCP_MEM_MAP_FIXED/NONBLOCK unsupported with non-host mem");
+        status = UCS_ERR_INVALID_PARAM;
+        goto out;
```

Comment:
again, can we move this check to UCT layer?
also, NONBLOCK is just a hint; can ignore it for cuda mem

Reply:
removed